src-tauri/src/commands.rs

```diff
@@ -532,3 +534,41 @@ pub async fn update_settings(data: SettingsPatch, handle: AppHandle) -> Result<S
     configure_tray_icon(&handle, &settings.tray_icon_theme)?;
     Ok(settings)
 }
+
+#[tauri::command(async)]
+pub async fn delete_instance(instance_id: i64, handle: AppHandle) -> Result<(), Error> {
+    debug!("Deleting instance {}", instance_id);
+    let app_state = handle.state::<AppState>();
+    let mut client = app_state.client.clone();
+    let pool = &app_state.get_pool();
+    if let Some(instance) = Instance::find_by_id(pool, instance_id).await? {
+        let instance_locations = Location::find_by_instance_id(pool, instance_id).await?;
+        if !instance_locations.is_empty() {
```

Comment:
To jest niepotrzebne jeżeli nie robisz żadnego `else`. Iteracja pustej listy po prostu nic nie zrobi.